ignite/metrics/vision/object_detection_average_precision_recall.py

```diff
@@ -0,0 +1,482 @@
+from typing import Callable, cast, Dict, List, Optional, Sequence, Tuple, Union
+
+import torch
+from typing_extensions import Literal
+
+from ignite.metrics import MetricGroup
+
+from ignite.metrics.mean_average_precision import _BaseAveragePrecision, _cat_and_agg_tensors
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+
+def coco_tensor_list_to_dict_list(
+    output: Tuple[
+        Union[List[torch.Tensor], List[Dict[str, torch.Tensor]]],
+        Union[List[torch.Tensor], List[Dict[str, torch.Tensor]]],
+    ]
+) -> Tuple[List[Dict[str, torch.Tensor]], List[Dict[str, torch.Tensor]]]:
+    """Convert either of output's `y_pred` or `y` from list of `(N, 6)` tensors to list of str-to-tensor dictionaries,
+    or keep them unchanged if they're already in the deisred format.
+
+    Input format is a `(N, 6)` or (`N, 5)` tensor which `N` is the number of predicted/target bounding boxes for the
+    image and the second dimension contains `(x1, y1, x2, y2, confidence, class)`/`(x1, y1, x2, y2, class[, iscrowd])`.
+    Output format is a str-to-tensor dictionary containing 'bbox' and `class` keys, plus `confidence` key for `y_pred`
+    and possibly `iscrowd` for `y`.
+
+    Args:
+        output: `(y_pred,y)` tuple whose members are either list of tensors or list of dicts.
+
+    Returns:
+        `(y_pred,y)` tuple whose members are list of str-to-tensor dictionaries.
+    """
+    y_pred, y = output
+    if len(y_pred) > 0 and isinstance(y_pred[0], torch.Tensor):
+        y_pred = [{"bbox": t[:, :4], "confidence": t[:, 4], "class": t[:, 5]} for t in cast(List[torch.Tensor], y_pred)]
+    if len(y) > 0 and isinstance(y[0], torch.Tensor):
+        if y[0].size(1) == 5:
+            y = [{"bbox": t[:, :4], "class": t[:, 4]} for t in cast(List[torch.Tensor], y)]
+        else:
+            y = [{"bbox": t[:, :4], "class": t[:, 4], "iscrowd": t[:, 5]} for t in cast(List[torch.Tensor], y)]
+    return cast(Tuple[List[Dict[str, torch.Tensor]], List[Dict[str, torch.Tensor]]], (y_pred, y))
+
+
+class ObjectDetectionAvgPrecisionRecall(Metric, _BaseAveragePrecision):
+    _tps: List[torch.Tensor]
+    _fps: List[torch.Tensor]
+    _scores: List[torch.Tensor]
+    _y_pred_labels: List[torch.Tensor]
+    _y_true_count: torch.Tensor
+    _num_classes: int
+
+    def __init__(
+        self,
+        iou_thresholds: Optional[Union[Sequence[float], torch.Tensor]] = None,
+        rec_thresholds: Optional[Union[Sequence[float], torch.Tensor]] = None,
+        num_classes: int = 91,
+        max_detections_per_image_per_class: int = 100,
+        area_range: Literal["small", "medium", "large", "all"] = "all",
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
```

Comment:
We also now have new arg `skip_unrolling` in Metric: https://github.com/pytorch/ignite/blob/680ac7f6dbd351a6c31f533f7b5be1b8b33242a3/ignite/metrics/metric.py#L236-L238